go/chat/boxer.go

```diff
@@ -202,16 +202,13 @@ func (b *Boxer) unboxV1(ctx context.Context, boxed chat1.MessageBoxed, encryptio
 	// compute the header hash
 	headerHash := b.hashV1(boxed.HeaderCiphertext.E)
 
+	// Whether the body is missing (deleted)
+	skipBodyVerification := (len(boxed.BodyCiphertext.E) == 0)
+
 	// decrypt body
 	// will remain empty if the body was deleted
 	var bodyVersioned chat1.BodyPlaintext
-	skipBodyVerification := false
-	if len(boxed.BodyCiphertext.E) == 0 {
-		if boxed.ServerHeader.SupersededBy == 0 {
-			return nil, NewPermanentUnboxingError(errors.New("empty body and not superseded in MessageBoxed"))
```

Comment:
Yep! So then a client downloads msg 2 and says "Wait a minute, this was deleted but not superseded! Fail to unbox."